rfqmsg/messages.go

```diff
@@ -182,3 +189,55 @@ type QuoteResponse interface {
 	// String returns a human-readable string representation of the message.
 	String() string
 }
+
+// Uint64FixedPoint is a fixed point record that can be used to encode/decode a
+// fixed point number with a uint64 value to/from a TLV stream.
+type Uint64FixedPoint rfqmath.FixedPoint[rfqmath.GoInt[uint64]]
+
+// NewUint64FixedPoint creates a new fixed point record with a uint64
+// value and a scale.
+func NewUint64FixedPoint(value uint64, scale int) Uint64FixedPoint {
+	return Uint64FixedPoint{
+		Value: rfqmath.NewGoInt(value),
+		Scale: scale,
+	}
+}
+
+// Record returns a TLV record that can be used to encode/decode an ID to/from a
+// TLV stream.
+//
+// NOTE: This is part of the tlv.RecordProducer interface.
+func (r *Uint64FixedPoint) Record() tlv.Record {
+	// We use a value of uint64 (8 bytes) and a scale of 1 byte.
+	const recordSize = 8 + 1
+
+	// Note that we set the type here as zero, as when used with a
+	// tlv.RecordT, the type param will be used as the type.
+	return tlv.MakeStaticRecord(
+		0, r, recordSize, Uint64FixedPointEncoder,
+		Uint64FixedPointDecoder,
+	)
+}
+
+// PriceQuote is a struct that holds the price quote by an oracle for a swap
+// between two assets.
```

Comment:
I think we should remove "by an oracle". Where a price quote comes from or what generates it is not relevant to what it is. We might store/retrieve via an archive. Maybe something like this:

```
// PriceQuote represents a price quote for an asset exchange operation.
//
// It contains the prices of both the input and output assets, expressed in
// asset units per BTC, using a fixed point representation. These prices can be
// considered as the exchange rates for swapping between the input and output
// assets. The quote is valid for a limited time and will expire.
```